lib/cdo/rake_utils.rb

```diff
@@ -100,9 +101,13 @@ def self.git_staged_changes?
     `git status --porcelain 2>/dev/null | egrep \"^(M|A|D)\" | wc -l`.strip.to_i > 0
   end
 
-  # Gets the commit hash for the given directory
-  def self.git_latest_commit_hash(dir)
-    `git log -n 1 --format="%H" -- #{dir}`.strip
+  # Gets a stable hash of the given directory's git-committed files.
+  # Uses a hash of the `git ls-tree` contents because a shallow-clone may not have the
+  # full revision history needed to find the original commit SHA.
+  def self.git_folder_hash(dir)
+    Dir.chdir(File.expand_path dir) do
```

Comment:
Style/NestedParenthesizedCalls: Add parentheses to nested method call `File.expand_path dir`.